### PR TITLE
flamp: init at 2.2.07

### PR DESCRIPTION
**Pull request by stteague**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/flamp/default.nix
Comment (natsukium) [inline pkgs/applications/radio/flamp/default.nix:26]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/radio/flamp/default.nix
Comment (natsukium) [inline pkgs/applications/radio/flamp/default.nix:27]:
```suggestion
    hash = "sha256-BnKL1iwaw1iJC5qcGCvdMr461n0BSYxk61dbAnhpz2c=";
```

File: pkgs/applications/radio/flamp/default.nix
Comment (natsukium) [inline pkgs/applications/radio/flamp/default.nix:30]:
```suggestion
  nativeBuildInputs = [
    autoreconfHook
    pkg-config
  ];
```
One item per line is preferred.

File: pkgs/applications/radio/flamp/default.nix
Comment (natsukium) [inline pkgs/applications/radio/flamp/default.nix:42]:
```suggestion
  ] ++ lib.optionals stdenv.isLinux [ 
    libpulseaudio
    alsa-lib
    udev
  ];
```

File: pkgs/applications/radio/flamp/default.nix
Comment (natsukium) [inline pkgs/applications/radio/flamp/default.nix:58]:
```suggestion
```
There is no need to define these here as they are the default behaviour.

File: maintainers/maintainer-list.nix
Comment (natsukium) [inline maintainers/maintainer-list.nix:1]: Would you please split this change into another commit?

Comment (stteague) [reply maintainers/maintainer-list.nix:1]: I rebased interactively for the first time and think I did it right.. I told it to squash to make fewer commits while I was at it. Thanks for your help.

File: pkgs/applications/radio/flamp/default.nix
Comment (drupol) [inline pkgs/applications/radio/flamp/default.nix:20]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/radio/flamp/default.nix
Comment (stteague) [reply pkgs/applications/radio/flamp/default.nix]: Hello @drupol - Thanks for the tip. I am new at anything Nix related. Was the change I just made correct? I was able to compile it on my system and test the package successfully after the change.

**Review by drupol — COMMENTED** [review]
LGTM now, but I'm unsure about the entries in `nativeBuildInputs` and `buildInputs`. I'll leave this to another maintainer.

Don't forget to squash your commits into one.

**Comment by natsukium** [comment]
It seems input is acceptable too.
Looking at other packages, maybe `gettext` should be in nativeBuildInputs.

**Comment by stteague** [comment]
> LGTM now, but I'm unsure about the entries in `nativeBuildInputs` and `buildInputs`. I'll leave this to another maintainer.
> 
> Don't forget to squash your commits into one.

I have done research, recompiling, and testing. The build inputs have been greatly reduced. That was my fault for pulling those in from the fldigi build without realizing flamp really doesn't need radio control and audio stuff since fldigi/flrig are the programs that do that stuff. Sorry, first time doing this.

On squashing commits: I was asked before to make the addition of myself as maintainer as a separate commit, so I made it two for now. Let me know if you want me to make it one, though, and I'll do that.

**Comment by natsukium** [comment]
LGTM, but one more thing.
Would you please change the commit message `Add self to maintainer-list.nix for flamp package` to `maintainers: add stteague`?

> On squashing commits: I was asked before to make the addition of myself as maintainer as a separate commit, so I made it two for now. Let me know if you want me to make it one, though, and I'll do that.

It may have left a commit in progress. The current two commits are correct.

**Comment by stteague** [comment]
> LGTM, but one more thing. Would you please change the commit message `Add self to maintainer-list.nix for flamp package` to `maintainers: add stteague`?
> 
> > On squashing commits: I was asked before to make the addition of myself as maintainer as a separate commit, so I made it two for now. Let me know if you want me to make it one, though, and I'll do that.
> 
> It may have left a commit in progress. The current two commits are correct.

I have amended the commit to have the correct description. Thanks for your help!

**Review by drupol — APPROVED** [review]

**Review by natsukium — APPROVED** [review]

**Review by drupol — APPROVED** [review]

**Pull request merged**
